doc: update partition mode config on up2

jsun26intel opened the pull request:
Update the using_partition_mode_on_up2.rst to accormodate the latest
configuration method for up2 partition mode.

Signed-off-by: Victor Sun <victor.sun@intel.com>

gvancuts commented on doc/tutorials/using_partition_mode_on_up2.rst:
```diff
@@ -167,133 +165,74 @@ Enable partition mode in ACRN hypervisor
    .. code-block:: none
-     :emphasize-lines: 3,39,44,49,56,72
-     :caption: hypervisor/partition/vm_description.c
+     :emphasize-lines: 5,9,17,21,25
```
I can see why it makes sense to highlight lines 9 and 21 (where the BDF collected above are used) but what is the meaning of the other lines highlighted?

I guess, looking at the current version of that doc, that the first BDF (`0:0.0`) is for the host bridge. Can we add this in the code like it used to be)? It would make things clearer.

I'm also assuming that we can use any BDF for the `vbdf.bits` as long as it hasn't been used yet. If true, some wording on this would be useful.

gvancuts commented on doc/tutorials/using_partition_mode_on_up2.rst:
```diff
@@ -167,133 +165,74 @@ Enable partition mode in ACRN hypervisor
+     };
+     ...
+
+#. Optionally, configure the ``VMx_CONFIG_OS_BOOTARGS`` kernel command line arguments
```
```suggestion
#. Optionally, configure the ``VMx_CONFIG_OS_BOOTARGS`` (where ``x`` is the VM number) kernel command line arguments
```

gvancuts commented on doc/tutorials/using_partition_mode_on_up2.rst:
```diff
@@ -167,133 +165,74 @@ Enable partition mode in ACRN hypervisor
+
+#. Optionally, configure the ``VMx_CONFIG_OS_BOOTARGS`` kernel command line arguments
+
+   The partiton mode configure infomation is located in header
```
```suggestion
   The partition mode configuration information is located in header
```

gvancuts commented on doc/tutorials/using_partition_mode_on_up2.rst:
```diff
@@ -167,133 +165,74 @@ Enable partition mode in ACRN hypervisor
+#. Optionally, configure the ``VMx_CONFIG_OS_BOOTARGS`` kernel command line arguments
+
+   The partiton mode configure infomation is located in header
+   file ``hypervisor/arch/x86/configs/apl-mrb/partition_config.h``.
```
Why is this file under `apl-mrb`? Could we promote it to `hypervisor/arch/x86/configs` if it's applicable to other platforms?

gvancuts commented on doc/tutorials/using_partition_mode_on_up2.rst:
```diff
@@ -167,133 +165,74 @@ Enable partition mode in ACRN hypervisor
+     #define VM0_CONFIG_OS_BOOTARGS		"root=/dev/sda3 rw rootwait noxsave maxcpus=2 nohpet console=hvc0 \
+						console=ttyS2 no_timer_check ignore_loglevel log_buf_len=16M \
+						consoleblank=0 tsc=reliable xapic_phys"
+
```
I would recommend to add a note here to indicate the the root device for `VM1` also is `/dev/sda3` since the USB controller is the only one seen in that VM.

gvancuts commented on doc/tutorials/using_partition_mode_on_up2.rst:
```diff
@@ -167,133 +165,74 @@ Enable partition mode in ACRN hypervisor
-     }
+     #define	VM0_CONFIGURED
+
+     #define VM0_CONFIG_NAME			"PRE-LAUNCHED VM1 for APL-MRB"
```
Could we change the names of the VMs if we move this file?

dbkinder commented:
@jsun26intel Please address @gvancuts comments and we'll re-review and merge for the 0.6 release

jsun26intel commented:
@dbkinder Sorry for the late reply, I am just back from CNY vacation. Please review my latest update.

jsun26intel commented:
@dbkinder @gvancuts 's comments should be addressed in my new push. 

I added a section of "Prepare VM configurations for UP2 partition mode" and addressed config folder and files which are specific for UP2 board. The section also addressed BDF for hostbridge and the meaning of some VMx_CONFIG_XXXX arguments.

dbkinder commented:
Since it's already the weekend in China, I'll merge this and submit a new PR to address Geoffroy's issues

dbkinder commented on doc/tutorials/using_partition_mode_on_up2.rst:
```diff
@@ -164,136 +162,142 @@ Enable partition mode in ACRN hypervisor
+
+   .. code-block:: none
+
+     $ cp hypervisor/arch/x86/configs/apl-mrb/* hypervisor/arch/x86/configs/up2/
```
... this works as written since there are no subdirectories in ``apl-mrb`` so I'll leave it alone

gvancuts replied:
We should at least create the up2 folder beforehand. That command will fail otherwise (this was in fact my primary motivation for suggesting this change)

gvancuts commented:
@dbkinder , could you also add a paragraph with the exact sequence of commands to be used to get this stuff compiled?

I would love to have @ttzeng review and validate this as well. I only checked the compilation phase, haven't attempted to run this.
